src/term/builtins.rs

```diff
@@ -94,11 +95,20 @@ impl Pattern {
   }
 
   fn encode_list(elements: Vec<Pattern>) -> Pattern {
-    let lnil = Pattern::Var(Some(Name::from(LNIL)));
+    let lnil = Pattern::Ctr(Name::from(LNIL), vec![]);
 
     elements.into_iter().rfold(lnil, |acc, mut nxt| {
       nxt.encode_builtins();
       Pattern::Ctr(Name::from(LCONS), vec![nxt, acc])
     })
   }
+
+  fn encode_str(str: &str) -> Pattern {
```

Comment:
Is this correct? the string is being encoded with list constructors?

Reply:
Definitely not